#! /usr/bin/env python

PACKAGE='blob_tracker'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("threshold", int_t, 0, "Fixed-level threshold for each image pixel.", 80, 0, 255)
gen.add("morph_kernel_size", int_t, 0, "Kernel size for morphological opening.", 20, 1, 100)
gen.add("center_marker_radius", int_t, 0, "Radius of center marker.", 20, 1, 100)
gen.add("drawn_line_thickness", int_t, 0, "Line thickness of drawn lines.", 10, 1, 40)
gen.add("draw_crosshairs",   bool_t,   0, "Draw crosshairs on display image.",  True)

# First string value is node name, used only for generating documentation
# Second string value ("ProcessImage") is name of class and generated
#    .h file, with "Config" added, so class ProcessImageConfig
exit(gen.generate(PACKAGE, "blob_tracker", "ProcessImage"))
